Tests/Runner/DataLoaderABSTest.php

```diff
@@ -0,0 +1,188 @@
+
+        // Delete file uploads
+        $options = new ListFilesOptions();
+        $options->setTags(["docker-bundle-test"]);
+        $files = $this->client->listFiles($options);
+        foreach ($files as $file) {
+            $this->client->deleteFile($file["id"]);
+        }
+    }
+
+    private function getNoDefaultBucketComponent()
+    {
+        return new Component([
+            'id' => 'docker-demo',
+            'data' => [
+                'definition' => [
+                    'type' => 'dockerhub',
+                    'uri' => 'keboola/docker-demo',
+                    'tag' => 'master'
+                ],
+
+            ]
+        ]);
+    }
+
+    public function testExecutorDefaultBucket()
+    {
+        $fs = new Filesystem();
+        $fs->dumpFile(
+            $this->workingDir->getDataDir() . '/out/tables/sliced.csv',
+            "id,text,row_number\n1,test,1\n1,test,2\n1,test,3"
+        );
+        $dataLoader = $this->getDataLoader([]);
+        $dataLoader->storeOutput();
+        self::assertTrue($this->client->tableExists('in.c-docker-demo-testConfig.sliced'));
+    }
+
+    public function testNoConfigDefaultBucketException()
+    {
+        self::expectException(UserException::class);
+        self::expectExceptionMessage('Configuration ID not set');
+        new DataLoader(
+            $this->client,
+            new NullLogger(),
+            $this->workingDir->getDataDir(),
+            [],
+            $this->getDefaultBucketComponent(),
+            new OutputFilter()
+        );
+    }
+
+    public function testExecutorInvalidOutputMapping()
+    {
+        $config = [
+            'input' => [
+                'tables' => [
+                    [
+                        'source' => 'in.c-docker-demo-testConfig.test',
+                    ],
+                ],
+            ],
+            'output' => [
+                'tables' => [
+                    [
+                        'source' => 'sliced.csv',
+                        'destination' => 'in.c-docker-demo-testConfig.out',
+                        // erroneous lines
+                        'primary_key' => 'col1',
+                        'incremental' => 1,
+                    ],
+                ],
+            ],
+        ];
+        $fs = new Filesystem();
+        $fs->dumpFile(
+            $this->workingDir->getDataDir() . '/out/tables/sliced.csv',
+            "id,text,row_number\n1,test,1\n1,test,2\n1,test,3"
+        );
+
+        $dataLoader = new DataLoader(
+            $this->client,
+            new NullLogger(),
+            $this->workingDir->getDataDir(),
+            $config,
+            $this->getNoDefaultBucketComponent(),
+            new OutputFilter()
+        );
+        self::expectException(UserException::class);
+        self::expectExceptionMessage('Failed to write manifest for table sliced.csv');
+        $dataLoader->storeOutput();
+    }
+
+    public function testLoadInputData()
```

Comment:
vsechny ostatni testy krome tohohle by daly dat pryc ne? s ABS vubec nekominíkují

Reply:
Jo, to je pravda